numerical_methods/roots.c

```diff
@@ -0,0 +1,166 @@
+#include <stdio.h>
+int value(double arr[], double m, double *e, int n)
+{
+    int b = n + 1;
+    int a = 0;
+    int c = 0;
+    double sum = 0;
+    double d = 1;
+    int i = n;
+    while (i >= 0)
+    {
+        while (c < a)
+        {
+            d = d * m;
+            c++;
+        }
+        sum = arr[i] * d + sum;
+        a++;
+        i--;
+    }
+    *e = sum;
+}
+int diff(double arr[], double m, double *c, int n)
+{
+    int a = n;
+    int i = 0;
+    double arr1[n];
+    while (i < n)
+    {
+        arr1[i] = arr[i + 1] * (a);
+        a--;
+        i++;
+    }
+    value(arr1, m, c, n);
+}
+int main()
+{
+    int n;
+    double p, q;
+    scanf("%d", &n);
+    int b = n + 1;
+    double arr[b];
+    double c, d, x, y;
+    double h;
+    int count = 0;
+    p = -100000;
+    for (int i = 0; i < b; i++)
+    {
+        scanf("%lf", &arr[i]);
+    }
+go:
+    if (count < n)
```

Comment:
Please document with proper Doxygen documentation all that's being done in the `main` function. It is recommended to split the code into functions as well.